src/workloads/execution/CumulativeWindows.yml

```diff
@@ -0,0 +1,155 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+
+Actors:
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Database: &db test
+    Threads: 1
+    CollectionCount: 1
+    # Small collection size to keep from spilling to disk during sorts.
```

Comment:
I wanted to find a way to avoid sorting, but it looks like a `sortBy` is still required even if there's an index defined on the `sortBy` field. Is that just a syntactic requirement? Does a sort get optimized away if the sort field is indexed?

Reply:
For example, here's a query with $sort then $group.  Explain shows the $sort is gone: instead the initial $cursor stage has an IXSCAN.
```
> db.c.explain().aggregate([ {$sort: {_id: 1}}, {$group: {_id: null, docs: {$push: "$$ROOT"}}} ])
{
        "explainVersion" : "2",
        "stages" : [
                {
                        "$cursor" : {
                                "queryPlanner" : {
                                        "namespace" : "test.c",
                                        "indexFilterSet" : false,
                                        "parsedQuery" : {

                                        },
                                        "queryHash" : "F44244FE",
                                        "planCacheKey" : "E15E9946",
                                        "maxIndexedOrSolutionsReached" : false,
                                        "maxIndexedAndSolutionsReached" : false,
                                        "maxScansToExplodeReached" : false,
                                        "winningPlan" : {
                                                "queryPlan" : {
                                                        "stage" : "FETCH",
                                                        "planNodeId" : 2,
                                                        "inputStage" : {
                                                                "stage" : "IXSCAN",
                                                                "planNodeId" : 1,
                                                                "keyPattern" : {
                                                                        "_id" : 1
                                                                },
                                                                "indexName" : "_id_",
                                                                "isMultiKey" : false,
                                                                "multiKeyPaths" : {
                                                                        "_id" : [ ]
                                                                },
                                                                "isUnique" : true,
                                                                "isSparse" : false,
                                                                "isPartial" : false,
                                                                "indexVersion" : 2,
                                                                "direction" : "forward",
                                                                "indexBounds" : {
                                                                        "_id" : [
                                                                                "[MinKey, MaxKey]"
                                                                        ]
                                                                }
                                                        }
                                                },
                                                "slotBasedPlan" : {
                                                        ...
                                                }
                                        },
                                        "rejectedPlans" : [ ]
                                }
                        }
                },
                {
                        "$group" : {
                                "_id" : {
                                        "$const" : null
                                },
                                "docs" : {
                                        "$push" : "$$ROOT"
                                }
                        }
                }
        ],
    ...
```

Reply:
Thanks @dpercy, in that case I will create an index on `t` since we already have other benchmarks that exercise the sort.